business_objects/record.py

```diff
@@ -88,6 +91,48 @@ def get_ids_of_manual_records_by_labeling_task(
     )
 
 
+def count_records_without_tokenization(project_id: str) -> int:
+    query = f"""
+    SELECT count(r.id) c
+    FROM ({get_records_without_tokenization(project_id, 0, True)}) r
+    """
+    return general.execute(query).first().c
+
+
+def get_attribute_data_with_doc_bins_of_records(
+    project_id: str, attribute_name: str
+) -> List[Any]:
+    query = f"""
+    SELECT rt.id, r."data" ->> '{attribute_name}' "attribute_data", rt.bytes
+    from record r
+    JOIN record_tokenized rt
+    ON  r.id = rt.record_id
+    WHERE r.project_id = '{project_id}'
+    AND rt.project_id = '{project_id}'
+    """
+    return general.execute_all(query)
+
+
+def update_bytes_of_record_tokenized(values: List[Dict[str, Any]]) -> None:
+    query = (
+        update(RecordTokenized)
+        .where(RecordTokenized.id == bindparam("_id"))
+        .values({"bytes": bindparam("bytes")})
+    )
+    general.execute(query, values)
+    general.flush()
+
+
+def update_columns_of_tokenized_records(rt_ids: List[str], attribute_name: str) -> None:
+    query = f"""
+    UPDATE record_tokenized 
+    SET columns = array_append(columns, '{attribute_name}')
+    WHERE id IN {rt_ids}
```

Comment:
does this work? The ids are a list which in turn should result in something like ['id1','id2'] --> SQL needs () imo.

Usually we do something like `rla_id_filter = "'" + "', '".join(rla_ids) + "'"` + `AND rla.id IN ({rla_id_filter})`

Reply:
Yeah you are right it would be a problem but rt_ids is a string parsed to fit the expected format

Reply:
then we should change the type in the function header so people don't get irritiated

Reply:
resolved